### PR TITLE
Indoor only filter

### PR DESCRIPTION
**Pull request by gitjul**
The end result with design will look like this:

Screenshot 2019-04-02 at 17 30 56

### CODE REVIEW

File: app/javascript/elm/src/Main.elm [hunk 1]
```diff
@@ -471,7 +484,7 @@ viewMobileFilters model =
     form [ Attr.class "filters-form" ]
         [ viewParameterFilter model.selectedParameter
         , Popup.viewPopup TogglePopupState SelectParameter model.isPopupExtended model.popup
-        , viewLocation model.location
+        , viewLocation model
```
Comment (AkkarinForest) [inline app/javascript/elm/src/Main.elm]: What do you think about passing `viewLocation model.location model.indoor`? It has a benefit of showing exactly which parts of the model are used in the function.

File: app/javascript/angular/code/filtersUtils.js [hunk 2]
```diff
@@ -105,3 +105,7 @@ export const clearLocation = (elmAction, params) => {
   elmAction.send(null);
   params.update({ data: { location: '' }});
 }
+
+export const toggleIndoor = (isIndoor, params) => {
+  params.update({ data: { indoor: isIndoor }});
+}
```
Comment (AkkarinForest) [inline app/javascript/angular/code/filtersUtils.js]: Does it have to be in filters utils even if it's only used in one place?

File: app/javascript/elm/tests/MainTests.elm [hunk 3]
```diff
@@ -417,6 +424,48 @@ crowdMapArea =
...
+                    |> Query.fromHtml
+                    |> Query.has [ Slc.attribute <| Attr.id "indoor-only-filter" ]
+        , test "checking checkbox triggers ToggleIndoor" <|
+            \_ ->
+                { defaultModel | page = Fixed }
+                    |> view
+                    |> Query.fromHtml
+                    |> Query.find [ Slc.attribute <| Attr.id "indoor-only-filter" ]
+                    |> Event.simulate (Event.check True)
+                    |> Event.expect (ToggleIndoor True)
```
Comment (AkkarinForest) [inline app/javascript/elm/tests/MainTests.elm]: this could be a `fuzz bool` test as well

File: public/partials/fixed_sessions_map.html [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+    </div>
+    <div ng-switch on="!!sensors.anySelected()">
+      <div ng-switch-when="true">
+        <h4>Heat Legend</h4>
+        <div class="section-divider" ng-include="versioner.path('/partials/heat_legend.html')"></div>
+      </div>
+    </div>
+  </div>
+</div>
```
Comment (AkkarinForest) [inline public/partials/fixed_sessions_map.html]: guess this file shouldn't be here :)

File: app/javascript/angular/code/controllers/_fixed_sessions_map_ctrl.js [hunk 5]
```diff
@@ -47,7 +47,7 @@ export const FixedSessionsMapCtrl = (
     const defaults = {
       sensorId,
       location: '',
-      indoorOnly: false,
+      indoor: false,
```
Comment (3v0k4) [inline app/javascript/angular/code/controllers/_fixed_sessions_map_ctrl.js]: is it an option to call this `isIndoor`?

File: app/javascript/elm/src/Main.elm [hunk 6]
```diff
@@ -48,6 +48,7 @@ type alias Model =
     , isCrowdMapOn : Bool
     , crowdMapResolution : Int
     , timeRange : TimeRange
+    , indoor : Bool
```
Comment (3v0k4) [inline app/javascript/elm/src/Main.elm]: have you considered `isIndoor` here?

File: app/javascript/elm/tests/MainTests.elm [hunk 7]
```diff
@@ -417,6 +424,48 @@ crowdMapArea =
...
+                    |> Query.fromHtml
+                    |> Query.find [ Slc.attribute <| Attr.id "indoor-only-filter" ]
+                    |> Query.has [ Slc.attribute <| Attr.checked indoorValue ]
+        , fuzz bool "ToggleIndoor triggers Ports.toggleIndoor with 'checked' value" <|
+            \indoorValue ->
+                { defaultModel | page = Fixed }
+                    |> update (ToggleIndoor indoorValue)
+                    |> Tuple.second
+                    |> Expect.equal (Ports.toggleIndoor indoorValue)
+        ]
```
Comment (3v0k4) [inline app/javascript/elm/tests/MainTests.elm]: Good job with the `fuzz`!

File: public/partials/fixed_sessions_map.html [hunk 8]
```diff
@@ -0,0 +1,41 @@
+<div id="sessions-tab">
```
Comment (3v0k4) [inline public/partials/fixed_sessions_map.html]: Is this file needed?

File: app/javascript/angular/code/filtersUtils.js (hunk 2 above)
Comment (gitjul) [reply app/javascript/angular/code/filtersUtils.js]: I guess not

File: app/javascript/elm/src/Main.elm (hunk 1 above)
Comment (gitjul) [reply app/javascript/elm/src/Main.elm]: I like it

File: app/javascript/elm/src/Main.elm (hunk 6 above)
Comment (gitjul) [reply app/javascript/elm/src/Main.elm]: yes! and now you mention it I'll actually do it :P

**Comment by gitjul** [comment]
In the spirit of smaller PRs I'm merging this keeping in mind that there are a few things to add on top of this before I submit the task for testing:

- [ ] add explanation why we're not displaying indoor sessions on the map on top of the overlay
- [ ] clear and disable profile names when displaying indoor sessions
- [ ] change checkbox into a toggle that Pina can style
